### PR TITLE
Feature: ResultCard component

### LINKED ISSUE #22: Create ResultCard component structure
Trello: https://trello.com/c/g5OZn1z8

The ResultCard has two variations:

### ResultCard Default State

Screen Shot 2020-06-05 at 10 50 53 AM

The ResultCard default state should output the following data from Airtable:
- Business Image - Does not exist in Airtable currently, read on.
  - Use image if available. If no image available, pull from fallback asset for the business category.
  - This data _does not_ currently exist in the database. Write conditionals to prepare for this, leverage a random unsplash image https://source.unsplash.com/random as a stub for the fallback.
- Business Name
- Business Category
- Business Blurb
  - This data _does not_ currently exist in the database. Write conditionals to prepare for this.
- Business Location
- Business Buttons
  - 1 - 2 buttons
  - Website URL button text dependent on business category
    - Entertainment: Learn more
    - Food and beverage: Order
    - Health and wellness: Learn more
    - Professional services: Contact
    - Retail: Shop
  - If no donate URL, the Website button takes full width.
  - If there is a donate URL provided, second button appears to the right, with button text “Donate”
- Report Link
  - mailto: link, use placeholder email for now.

### ResultCard Basic

Screen Shot 2020-06-05 at 11 00 29 AM

This variation does not utilize a Business image, but instead leverages a background color.

- Business Name
- Business Location
- Business Blurb
  - This data _does not_ currently exist in the database. Write conditionals to prepare for this.
- Business Button
  - 1 Button
  - Website URL. Button text "Website"
- Report Link
  - mailto: link, use placeholder email for now.

### PR DESCRIPTION
**Pull request by chancestrickland**
This PR creates a new `ResultCard` component as well as more generic composable `Card` components. Closes #22.

**IMPORTANT:** There are edits to the homepage that should not be merged. These are intended to make it quick to review these components locally.

Component API:

```js
<ResultCard
  // assumes these come from a list somewhere, most likely
  businessCategory="foodAndBeverage"
  businessName="Dad's Donuts"
  businessDescription="Blah blah blaaaah"
  businessLocation="Atlanta, GA"
  businessUrl="https://dobusiness.com"
  donationUrl="https://donatemoney.com"
  // overrides the default fallback image
  imageSrc="https://imagesrc.jpg"
  imageAlt="cool image you got there"
/>
```

### Screenshots / video of changes

image

## Steps to test

Run locally and check out the homepage.

### CODE REVIEW

**Comment by chancestrickland** [comment]
OH and BTW, I added eslint dependencies because...well, there was no linting and I was making lots of mistakes. I can put this in another PR if it's too cluttery here, or if someone else is handling this elsewhere. Just let me know.

**Comment by magnificode** [comment]
@chancestrickland this is looking **solid** - Is there any chance we can incorporate the work @kkemple did here https://github.com/Rebuild-Black-Business/RBB-Website/pull/35 - Just want to make sure we get proper schema markup baked into the ResultCard component for businesses.

Also very much appreciate you adding that linting. Need to have that in sooner rather than later for suuure.

**Comment by racedale** [comment]
@chancestrickland some linting was merged to master today, make sure to merge master in!

File: src/components/ResultCard.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+const ResultCard = forwardRef(
+  (
+    {
+      children,
+      imageSrc,
+      imageAlt,
+      businessCategory,
+      businessName,
+      businessDescription,
+      businessLocation,
```
Comment (ChrisBrownie55) [inline src/components/ResultCard.js:68]: It seems a tad redundant to have `business` in front of each word. Some options to make the code a little less verbose could be an **object with all of the business stuff** in it (i.e. `business.category`, `business.name`) and that could _also apply to things like the image_ (i.e. `image.src`, `image.alt`).

Or another option could just be having the unique names be without the first part (i.e. `src`, `alt`, `category`, `name`) but that could be confusing.

Comment (chancestrickland) [reply src/components/ResultCard.js:68]: Prefer to stay away from object literal props as a general rule if I can, but happy to drop the business prefix.

**Comment by chancestrickland** [comment]
@magnificode I incorporated some of @kkemple's patterns, though it seems like we still need to flesh out the categories a bit. Left some todos in place but let me know those are going to be handled. I imagine we'll pull something from Airtable in whatever context these components are used rather than the component itself, so we can probably revisit this after merging. 

@racedale I updated and merged with master.

File: package.json [hunk 2]
```diff
@@ -47,16 +47,25 @@
...
     "babel-jest": "^26.0.1",
     "babel-preset-gatsby": "^0.4.8",
     "cypress": "^4.7.0",
     "cypress-axe": "^0.8.1",
+    "eslint": "6.x",
     "eslint-config-prettier": "^6.11.0",
     "eslint-config-react-app": "^5.2.1",
     "eslint-plugin-chai-friendly": "^0.6.0",
     "eslint-plugin-cypress": "^2.11.1",
+    "eslint-plugin-flowtype": "4.x",
```
Comment (chancestrickland) [inline package.json:63]: Just a note, all the new deps here are all peer deps for `eslint-config-react-app`.

**Comment by magnificode** [comment]
> @magnificode I incorporated some of @kkemple's patterns, though it seems like we still need to flesh out the categories a bit. Left some todos in place but let me know those are going to be handled. I imagine we'll pull something from Airtable in whatever context these components are used rather than the component itself, so we can probably revisit this after merging.
> 
> @racedale I updated and merged with master.

Yeah no worries at all, appreciate you keeping things flexible there and adding to-dos. This will continue to be a bit fluid as designs start rolling in. Very much appreciate the work here @chancestrickland !

**Comment by magnificode** [comment]
@chancestrickland - feel free to remove the examples on the home page for now! If you wanted to turn the snippet you have in your original PR message into commented documentation in the Component file itself that would be awesome! Just to get a little usage example in the file itself. 

Ping me when you've got that handled and I'll merge this in. Thank you again for the great work!

File: src/components/Button.js [hunk 3]
```diff
@@ -38,12 +39,20 @@ function getButtonStyles(theme, variant) {
...
   const buttonStyles = getButtonStyles(theme, variant);
-  return <ChakraButton {...buttonStyles} {...props} />;
-}
+  return <ChakraButton {...buttonStyles} ref={ref} {...props} />;
+});
+
+Button.displayName = 'Button';
+Button.propTypes = {
+  variant: PropTypes.oneOf(['cta', 'primary', 'secondary']),
+};
```
Comment (jakeleboeuf) [inline src/components/Button.js:55]: I've been in Typescript world for so long I forgot about PropTypes 😍

Comment (chancestrickland) [reply src/components/Button.js:55]: OMG SAME

**Comment by chancestrickland** [comment]
@magnificode Removed the homepage stuff, added that snippet in a JSDoc block. Holler if you need anything else to 🚢

**Pull request merged**